### PR TITLE
Fix RestOperator and SpreadOperator for eslint2

### PR DESCRIPTION
**Pull request by danez**
Correctly transform RestOperator to ExperimentalRestProperty.

### CODE REVIEW

File: babylon-to-espree/toAST.js [hunk 1]
```diff
@@ -165,7 +165,7 @@ var astTransformVisitor = {
     }
 
     if (path.isRestProperty() || path.isSpreadProperty()) {
-      node.type = "SpreadProperty";
+      node.type = "ExperimentalRestProperty";
```
Comment (hzoo) [inline babylon-to-espree/toAST.js]: Do we need to make a separate one for `ExperimentalSpreadProperty`?

**Comment by hzoo** [comment]
Cool!

File: test/babel-eslint.js [hunk 2]
```diff
@@ -36,7 +36,7 @@ function lookup(obj, keypath, backwardsDepth) {
   if (!keypath) { return obj; }
 
   return keypath.split(".").slice(0, -1 * backwardsDepth)
-  .reduce(function (base, segment) { base && base[segment], obj });
+  .reduce(function (base, segment) { return base && base[segment], obj });
```
Comment (hzoo) [inline test/babel-eslint.js]: Thanks - I put an arrow function and saw that node 0.10 was failing haha

File: babylon-to-espree/toAST.js (hunk 1 above)
Comment (danez) [inline babylon-to-espree/toAST.js]: A separate on of what? You mean splitting Rest and Spread? I just tested and it seems that for SpreadProperty there is nothing todo here at all. `key`and `value` are not used in espree (anymore?).  Gonna push a change for that.

Comment (hzoo) [inline babylon-to-espree/toAST.js]: Yeah I meant the `type`.

https://github.com/eslint/espree/blob/e4b561c2dd543cce0869a077c0fc158e1d5b1990/lib/ast-node-types.js#L59-L60

File: babylon-to-espree/toAST.js [hunk 3]
```diff
@@ -164,9 +164,8 @@ var astTransformVisitor = {
       delete node.typeParameters;
     }
 
-    if (path.isRestProperty() || path.isSpreadProperty()) {
-      node.type = "SpreadProperty";
-      node.key = node.value = node.argument;
+    if (path.isRestProperty()) {
+      node.type = "ExperimentalRestProperty";
```
Comment (hzoo) [inline babylon-to-espree/toAST.js]: Basically do the same for `isSpreadProperty`?

Comment (danez) [inline babylon-to-espree/toAST.js]: You are right, I forgot that there is `ExperimentalSpread...` and `ExperimentalRest...`

**Pull request merged**